php-cs-fixer

```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env php
 <?php
 
+declare(strict_types=1);
```

Comment:
Idea was to compile even on 5.6 (supported by PHP CS Fixer v2) - now it does not.
We should align which PHP version this file should support and have CI action checking that.

remember that binary .phar file can be downloaded and executed on any PHP version, and nice msg should be displayed - i value that more over non-abaility to add this file under SCA

Reply:
I hope https://stitcher.io/blog/php-version-stats-january-2024 can justify this over the CI improvement value.

Reply:
it doesn't.
it still doesn't ensure that we put proper, clear communication to end-user.

imagine file would be valid only for versions we allow in composer.json - 7.4+.
that means crash msg for ~7% users, taking numbers from page you shared.

Reply:
No! Are you aware of https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8229#discussion_r1780948706 discussion?

I checked this using https://3v4l.org/XpoBj#v5.6.40 again.

Result: PHP 5.6 is already not supported on master and PHP 7.0 support is kept after this PR.

So all right?

Reply:
I can confirm we already lost compatibility with PHP 5.6:

```shell
docker run -it --rm -v $(pwd):/code php:5.6 php -l /code/php-cs-fixer

Parse error: syntax error, unexpected '(' in /code/php-cs-fixer on line 57
Errors parsing /code/php-cs-fixer
```

It seems like self-invoking functions (IIFE) are supported only on 7.0+ 🙂. I've fixed it in #8235, let's finish that one first, so we can continue with BC-promise check here.